src/common/utils/DelayedJobs.ts

```diff
@@ -0,0 +1,70 @@
+export class DelayedJobs {
+  private _delayedJobs = new Map<string, any>();
+
+  public cancel(tag: string): boolean {
+    const job = this._delayedJobs.get(tag);
+    this._delayedJobs.delete(tag);
+    if (!job) return false;
+    clearTimeout(job.timeoutId);
+    job.setCanceled();
+    return true;
+  }
+
+  public instantlyRunAllOf(kind?: number): Promise<any> {
+    const promises = [];
+    const jobs = this.getJobsOfKind(kind);
+    for (const job of jobs) {
+      clearTimeout(job.timeoutId);
+      promises.push(job.setComplete());
+    }
+    return Promise.allSettled(promises);
+  }
+
+  public schedule(
+    delayMs: number,
+    tag: string,
+    kind: number,
+    action: () => Promise<void>
+  ): Promise<void> {
+    return new Promise<void>((resolve, reject) =>{
```

Comment:
I understand your reasoning in favor of async/await use, @Dotneteer . But unfortunately I cannot take this Promise out as it enables the schedule method clients to await for the scheduled action right away. Currently, this feature lies in the basis of the editors' 'document-is-dirty' state tracking.